lib/huginn_acumen_product_agent/concerns/acumen_product_query_concern.rb

```diff
@@ -375,9 +375,11 @@ def process_products_and_variants(products, variants, links, physical_formats, d
         links.each do |link|
             from_id = link['from_id']
             to_id = link['to_id']
-            variant = variants_map[to_id]
-            variant['isDefault'] = false
-            products_map[from_id]['model'].push(*variant['model'])
+            if variants_map.key?(to_id)
```

Comment:
@Jhalter5Stones why is the key missing from the map in the first place?

Reply:
@jspizziri Some products in Acumen have linked products where both links have an alt format of 1 rather than one of them as 0 and the other as 1. So I believe the key is missing because an extra link was created because both of the products had alt format 1. On all products that are syncing correctly I am seeing at least one alt format (boolean) as 0 where as on the products that are breaking in the sync have all alt formats (boolean) set to 1

Reply:
@mrogers-5s , if this makes sense to you I'm ready to sign off.

Reply:
@Jhalter5Stones @jspizziri 

I'm not sure Justin's _explanation_ is correct, but I think the logic here is.

Looking at the code, I _think_ what's actually happening is that the Linked Products query is returning records for product IDs that no longer exist.

[This method](https://github.com/5-stones/huginn_acumen_product_agent/blob/22c8ed3b6435fdbdf6b12ec8bfdc96e3872df564/lib/huginn_acumen_product_agent/concerns/acumen_product_query_concern.rb#L21) gets product links based on the "master" product ID. (`from_id`).

For each result where `alt_format = true`, we pull the `to_id` to get a list of variant IDs, which we pass [here](https://github.com/5-stones/huginn_acumen_product_agent/blob/22c8ed3b6435fdbdf6b12ec8bfdc96e3872df564/lib/huginn_acumen_product_agent/concerns/acumen_product_query_concern.rb#L21) to get the `Inv_Product` record.

Then in the method above that Justin modified, we [map the results](https://github.com/5-stones/huginn_acumen_product_agent/blob/master/lib/huginn_acumen_product_agent/concerns/acumen_product_query_concern.rb#L373) of that query to `variants_map`.

The reason the `key` doesn't exist, is because the product ID we pulled from the link doesn't have a matching product ID on the `Inv_Product` table.

I think it's just a coincidence that this is happening for records where all the links are alternate formats.